ZRC: Add Readme docs to React Components stories

eatyourgreens opened the pull request:
Storybook v6 deprecated `addon-notes` in favour of the Docs page for each component story. This PR replaces the `notes` parameter with the `docs` parameter for all the ZRC stories. This should add each component's Readme to its Docs page in the storybook.
![Screenshot of the Favourites Button story, showing the component's Readme on its Docs page.](https://user-images.githubusercontent.com/59547/104301314-68e11400-54bf-11eb-8f6e-6313c2a572e8.png)

Package:
lib-react-components

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

ErikOstlund commented:
@eatyourgreens, component README.md files are appearing in Storybook under the `Docs` tab. Looks like the `docs` parameter is working, but will you advise on this?...

I noticed Tooltip and ZooFooter components have story files but do not appear in Storybook (local dev).
Files:
![Screen Shot 2021-01-12 at 2 37 31 PM](https://user-images.githubusercontent.com/9533007/104370306-49101700-54e4-11eb-8709-eb86ac7c7615.png)

Storybook:
![Screen Shot 2021-01-12 at 2 42 16 PM](https://user-images.githubusercontent.com/9533007/104370405-6a710300-54e4-11eb-9302-ac048378059c.png)

Also to note: In Storybook, there is a Triangle component, but I don't see a triangle component in our code.

Are these known issues?

eatyourgreens commented:
@ErikOstlund thanks for taking a look. That sounds like a bug from the storybook upgrade. I had a look at the tooltip and the footer, but I don’t see an obvious reason why they aren’t appearing.

eatyourgreens commented:
Looking at the deployed storybook, ZooFooter and Tooltip are navigation categories now, because they’re the only two components that have other stories in directories underneath them. Maybe the sidebar structure needs re-organising?
https://zooniverse.github.io/front-end-monorepo/@zooniverse/react-components/index.html?path=/story/zoofooter--light-with-admin

eatyourgreens commented:
Weird that ZooFooter gets moved to the top of the sidebar.

eatyourgreens commented:
The Storybook docs have [a helpful reference for the sidebar hierarchy](https://storybook.js.org/docs/react/writing-stories/naming-components-and-hierarchy). I'm going to push some changes to the ZRC story titles, after reading that.

eatyourgreens commented:
@ErikOstlund I've changed the name of the `AdminCheckbox` story and rewritten the Tooltip stories as CSF, after reading the docs for [single story hoisting](https://storybook.js.org/docs/react/writing-stories/naming-components-and-hierarchy#single-story-hoisting).

ErikOstlund reviewed (approved):
Looks good. All story components display as expected. README files display under the Docs tab as expected. Well done!
